addons/payment/controllers/portal.py

```diff
@@ -175,6 +175,19 @@ def payment_method(self, **kwargs):
         }
         return request.render('payment.payment_methods', rendering_context)
 
+    @http.route('/my/payment_method/archive', type='json', auth='user')
+    def archive_token(self, token_id):
+        user_partner = request.env.user.partner_id
+        token_sudo = request.env['payment.token'].sudo().search([
+            ('id', '=', token_id),
+            ('partner_id', 'in', [user_partner.id, user_partner.sudo().commercial_partner_id.id])
+        ])
+
+        if token_sudo:
+            token_sudo.sudo().write({'active': False})
```

Comment:
Again, the `'_sudo'` is helpful.
```suggestion
            token_sudo.write({'active': False})
```